### PR TITLE
Onboarding: Various improvements

### PR DESCRIPTION
**Pull request by emilylaguna**
| 📘 Project: #500 |
|:---:|

This includes some various improvements and bug fixes:
- Adds back button to the onboarding views
- Updates the login header to:
  - Show larger images when SSO is disabled
  - Show random podcasts from the users database
- Styles the not now button
- Styles the back button for the plus landing
- Increases line spacing of the plus terms / privacy text
- No longer allows the user to swipe back to the plus purchase view
- Fixing some issues with the motion jumping
- Shows a close button if the plus landing view is the root view

## To test
1. Enable the FeatureFlag
2. Launch the app
3. Tap the Podcasts tab
4. Tap the Create Folder button
5. ✅ The login header looks good, the not now is styled
6. Tap the Log in button
7. ✅ Verify it has a back button that is white
8. Tap unlock all and subscribe
9. ✅ Verify you can't swipe back from the welcome view anymore
10. Tap on Import
11. ✅ Verify there is a back button
12. Tap on an import item
13. ✅ Verify there is a back button
14. Tap on Profile > Cog > Appearance
15. Select a theme that is locked
16. ✅ Verify you see the plus view with a close button

## Checklist

- [ ] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

**Comment by pocketcasts** [comment]
### SwiftLint found issues

#### Warnings

File | Line | Reason |
| --- | ----- | ----- |
Motion.swift | 45 | Lines should not have trailing whitespace. (trailing_whitespace)


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: podcasts/End of Year/Stories/Views/PodcastCover.swift [hunk 1]
```diff
@@ -9,21 +9,38 @@ struct PodcastCover: View {
...
+    }
+
+    private var rectangleColor: Color? {
+        var rectangleColor: Color?
+
+        if let viewBackgroundStyle {
+            rectangleColor = AppTheme.color(for: viewBackgroundStyle)
+        }
+
+        return rectangleColor
```
Comment (leandroalonso) [inline podcasts/End of Year/Stories/Views/PodcastCover.swift]:
```suggestion
        if let viewBackgroundStyle {
            return AppTheme.color(for: viewBackgroundStyle)
        }

        return nil
```

File: podcasts/End of Year/Stories/Views/PodcastCover.swift [hunk 2]
```diff
@@ -9,21 +9,38 @@ struct PodcastCover: View {
...
     }
 
     var body: some View {
         ZStack {
             Group {
                 if big {
                     Rectangle()
+                        .foregroundColor(rectangleColor)
```
Comment (leandroalonso) [inline podcasts/End of Year/Stories/Views/PodcastCover.swift:37]: When the color is nil the `Rectangle` will behave the same as if it was without this property?

File: podcasts/PodcastListViewController.swift [hunk 3]
```diff
@@ -220,22 +220,22 @@ class PodcastListViewController: PCViewController, UIGestureRecognizerDelegate,
...
-            let navigationController = OnboardingNavigationViewController(rootViewController: controller)
-            viewModel.navigationController = navigationController
-
-            self.present(navigationController, animated: true)
-
-//            let coordinator = LoginCoordinator()
-//            let view = LoginLandingView(coordinator: coordinator)
-//            let controller = LoginLandingHostingController(rootView: view.setupDefaultEnvironment(), coordinator: coordinator)
-//
+//            let viewModel = WelcomeViewModel(displayType: .plus)
```
Comment (leandroalonso) [inline podcasts/PodcastListViewController.swift:223]: Given we're merging to `trunk` shall we add a TODO here? Or do you think it's not necessary?

**Review by leandroalonso — APPROVED** [review]

File: podcasts/PodcastListViewController.swift (hunk 3 above)
Comment (emilylaguna) [reply podcasts/PodcastListViewController.swift:223]: No todo necessary I am working on the real flow.

**Pull request merged**